pkg/reconciler/trigger/trigger.go

```diff
@@ -0,0 +1,378 @@
+)
+
+const (
+	// ReconcilerName is the name of the reconciler
+	ReconcilerName = "Subscriptions"
+
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
+	controllerAgentName = "trigger-controller"
+
+	finalizerName = controllerAgentName
+
+	// Name of the corev1.Events emitted from the reconciliation process.
+	triggerReconciled         = "TriggerReconciled"
+	triggerReconcileFailed    = "TriggerReconcileFailed"
+	triggerUpdateStatusFailed = "TriggerUpdateStatusFailed"
+	subscriptionDeleteFailed  = "SubscriptionDeleteFailed"
+	subscriptionCreateFailed  = "SubscriptionCreateFailed"
+)
+
+type Reconciler struct {
+	*reconciler.Base
+
+	triggerLister      listers.TriggerLister
+	channelLister      listers.ChannelLister
+	subscriptionLister listers.SubscriptionLister
+	brokerLister       listers.BrokerLister
+	serviceLister      corev1listers.ServiceLister
+}
+
+// Check that our Reconciler implements controller.Reconciler
+var _ controller.Reconciler = (*Reconciler)(nil)
+
+// NewController initializes the controller and is called by the generated code
+// Registers event handlers to enqueue events
+func NewController(
+	opt reconciler.Options,
+	triggerInformer eventinginformers.TriggerInformer,
+	channelInformer eventinginformers.ChannelInformer,
+	subscriptionInformer eventinginformers.SubscriptionInformer,
+	brokerInformer eventinginformers.BrokerInformer,
+	serviceInformer corev1informers.ServiceInformer,
+) *controller.Impl {
+
+	r := &Reconciler{
+		Base:               reconciler.NewBase(opt, controllerAgentName),
+		triggerLister:      triggerInformer.Lister(),
+		channelLister:      channelInformer.Lister(),
+		subscriptionLister: subscriptionInformer.Lister(),
+		brokerLister:       brokerInformer.Lister(),
+		serviceLister:      serviceInformer.Lister(),
+	}
+	impl := controller.NewImpl(r, r.Logger, ReconcilerName, reconciler.MustNewStatsReporter(ReconcilerName, r.Logger))
+
+	r.Logger.Info("Setting up event handlers")
+	triggerInformer.Informer().AddEventHandler(reconciler.Handler(impl.Enqueue))
+
+	channelInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Trigger")),
+		Handler:    reconciler.Handler(impl.EnqueueControllerOf),
+	})
+
+	subscriptionInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Trigger")),
+		Handler:    reconciler.Handler(impl.EnqueueControllerOf),
+	})
+	return impl
+}
+
+// Reconcile compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Trigger resource
+// with the current status of the resource.
+func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
```

Comment:
Save the key into the logger:

```golang
ctx = logging.WithLogger(ctx, r.logger.With(zap.String("key", key)))
```

Then all subsequent log lines don't need to include the key.

Reply:
done, just had to desugar in the With call?

Reply:
Eventing's `logging.WithLogger`/`logging.FromContext` use a desugared logger. If you want to switch to sugared change the import to use pkg's logging library instead. 

My preference is desugar as it is harder to get things wrong by accident.